### PR TITLE
[dart] mark json_serializable as experimental

### PR DESCRIPTION
**Pull request by agilob**
this generator hasn't been released yet and I don't think it is ready to be released. upcoming changes for dart 2.12 and j_s 4.x are breaking changes, so I'd better hide it from users but keep the code than release a breaking change without fallback after 5.1.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.
@kuhnroyal

### CODE REVIEW

**Comment by agilob** [comment]
Some golang error..

```
Total time: 19.777 secs
unrecognized import path "gopkg.in/yaml.v3": reading https://gopkg.in/yaml.v3?go-get=1: 502 Bad Gateway
	server response: Cannot obtain refs from GitHub: cannot talk to GitHub: Get https://github.com/go-yaml/yaml.git/info/refs?service=git-upload-pack: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2.1:exec (go-get-testify) on project GoPetstore: Command execution failed. Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] 
```

**Comment by agilob** [comment]
Unrelated error again

```
Exception when calling PetApi#getPetById
Status code: 404
Reason: {"code":1,"type":"error","message":"Pet not found"}
Response headers: {Transfer-Encoding=[chunked], Server=[Jetty(9.2.9.v20150224)], Access-Control-Allow-Origin=[*], Access-Control-Allow-Methods=[GET, POST, DELETE, PUT], Date=[Tue, 16 Mar 2021 07:53:08 GMT], Access-Control-Allow-Headers=[Content-Type, api_key, Authorization], Content-Type=[application/json]}
DATE: 2011-01-18T00:00:00.000+00:00
DATE: 2011-01-18T00:00:00.000+00:00
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[ERROR] Tests run: 13, Failures: 0, Errors: 1, Skipped: 1, Time elapsed: 1.708 s <<< FAILURE! - in org.openapitools.client.api.PetApiTest
[ERROR] org.openapitools.client.api.PetApiTest.testFindPetsByStatus  Time elapsed: 0.692 s  <<< ERROR!
org.openapitools.client.ApiException: Server Error
	at org.openapitools.client.api.PetApiTest.testFindPetsByStatus(PetApiTest.java:298)

[ERROR] Errors: 
[ERROR]   PetApiTest.testFindPetsByStatus:298 » Api Server Error
[ERROR] Tests run: 252, Failures: 0, Errors: 1, Skipped: 6
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M4:test (default-test) on project petstore-okhttp-gson: There are test failures.
```

**Review by kuhnroyal — APPROVED** [review]
Yea, I agree. Probably have to go with a new generator and make it default by 6.0.

**Comment by agilob** [comment]
I mean, it can go as new generator for 5.2. I'm more worried that json_serializable team added new required parameter  to JsonKey annotation and changed name of another param, updating it would be a breaking change, so no point in releasing in 5.1 just to break it in 5.2..

**Comment by kuhnroyal** [comment]
Dart 2.12 support is breaking anyways for a lot of libraries. I would argue we should break in 5.1 - In 3 month nobody is gonna use these old Dart generators anymore. Dart moves way too fast.

**Comment by agilob** [comment]
pls merge before 5.1 @wing328

**Comment by wing328** [comment]
Suggestion: What about marking `json_serializable` as experimental and state that it's subject to breaking changes without further notice?

**Comment by agilob** [comment]
You're the boss and I'm ok with that. serializer would still work but wouldnt be documented, making it experimental is fine with me, but I will introduce breaking changes around mid-April, didnt want it to be surprise for users.

**Comment by wing328** [comment]
I just happen to contribute a bit more than the others :)

File: docs/generators/dart.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ These options may be applied as additional-properties (cli) or configOptions (pl
 |pubLibrary|Library name in generated code| |null|
 |pubName|Name in generated pubspec| |null|
 |pubVersion|Version in generated pubspec| |null|
-|serializationLibrary|Specify serialization library|<dl><dt>**native_serialization**</dt><dd>Use native serializer, backwards compatible</dd><dt>**json_serializable**</dt><dd>Use json_serializable</dd></dl>|native_serialization|
+|serializationLibrary|Specify serialization library|<dl><dt>**native_serialization**</dt><dd>Use native serializer, backwards compatible</dd><dt>**json_serializable**</dt><dd>Use json_serializable. Experimental and subject to breaking changes in minor releases</dd></dl>|native_serialization|
```
Comment (wing328) [inline docs/generators/dart.md:22]: > Experimental and subject to breaking changes in minor releases

What about `Experimental and subject to breaking changes without further notice` as you probably want to do breaking changes in patch releases as well (e.g. 5.1.0 => 5.1.1) ?

Comment (agilob) [reply docs/generators/dart.md:22]: done

**Comment by agilob** [comment]
>fatal: unable to access 'https://github.com/OpenAPITools/openapi-generator.git/': Could not resolve host: github.com

**Pull request merged**